### ISSUE TITLE
Would you accept a PR that fixes the 1.6 warnings?

### ISSUE THREAD

**Issue by pragdave** [elixir-mongo/mongodb#188]
~~~
==> mongodb
Compiling 28 files (.ex)
warning: function slice/1 required by protocol Enumerable is not implemented (in module Enumerable.Mongo.Cursor)
  lib/mongo/cursor.ex:20

warning: GenEvent.notify/2 is deprecated, use one of the alternatives described in the documentation for the GenEvent module
  lib/mongo/events.ex:4

warning: function slice/1 required by protocol Enumerable is not implemented (in module Enumerable.Mongo.AggregationCursor)
  lib/mongo/cursor.ex:122

warning: function slice/1 required by protocol Enumerable is not implemented (in module Enumerable.Mongo.SinglyCursor)
  lib/mongo/cursor.ex:203

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/monitor.ex:47

warning: Kernel.to_char_list/1 is deprecated, use Kernel.to_charlist/1
  lib/mongo/protocol.ex:82

warning: Kernel.to_char_list/1 is deprecated, use Kernel.to_charlist/1
  lib/mongo/protocol.ex:93

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:72

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:107

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:124

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:185

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:190

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:212

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:217

warning: variable "self" does not exist and is being expanded to "self()", please use parentheses to remove the ambiguity or change the variable name
  lib/mongo/topology.ex:226

warning: function check_topology/1 is unused
  lib/mongo/topology_description.ex:546


warning: function invalid_set_name?/1 is unused
  lib/mongo/topology_description.ex:555

warning: variable "key" is unused
  lib/mongo/topology.ex:141

warning: variable "ret" is unused
  lib/mongo/topology_description.ex:259

warning: GenEvent.stream/2 is deprecated, use one of the alternatives described in the documentation for the GenEvent module
  lib/mongo.ex:755
~~~

**Comment by ankhers** [comment]
I would appreciate any help you would be willing to offer.

**Comment by pragdave** [comment]
I'm on it.


On Thu, Jan 25, 2018 at 9:18am, Justin Wood < notifications@github.com [notifications@github.com] > wrote:
I would appreciate any help you would be willing to offer.

**Comment by ankhers** [comment]
Mostly resolved in #191
